libraries/networking/src/udt/Connection.cpp

```diff
@@ -589,6 +619,26 @@ void Connection::processNAK(std::unique_ptr<ControlPacket> controlPacket) {
     _stats.record(ConnectionStats::Stats::ReceivedNAK);
 }
 
+void Connection::processHandshake(std::unique_ptr<ControlPacket> controlPacket) {
+    // server sent us a handshake - we need to assume this means state should be reset
+    resetReceiveState();
```

Comment:
As we talked about, we should ignore the handshake if we have not received any other packets. This will fix the case that your RTT is greater than the handshake send interval time (100ms).